File: legacy/decompiled/sources/com/nemo/caideng/util/HexStringUtil.java
package com.nemo.caideng.util;

/* JADX INFO: loaded from: classes.dex */
public class HexStringUtil {
    public static byte[] hexToByteArray(String str) {
        byte[] bArr = new byte[str.length() / 2];
        int length = str.length();
        int i = 0;
        while (i <= length - 1) {
            int i2 = i + 2;
            bArr[i / 2] = (byte) Integer.parseInt(str.substring(i, i2), 16);
            i = i2;
        }
        return bArr;
    }

    public static String byteArrayToHex(byte[] bArr) {
        StringBuilder sb = new StringBuilder();
        int length = bArr.length;
        for (int i = 0; i <= length - 1; i++) {
            char cForDigit = Character.forDigit((bArr[i] >> 4) & 15, 16);
            char cForDigit2 = Character.forDigit(bArr[i] & 15, 16);
            sb.append(cForDigit);
            sb.append(cForDigit2);
        }
        return sb.toString();
    }
}

File: app/src/main/java/com/mualanimarine/thalopilot/util/HexStringUtil.kt
package com.mualanimarine.thalopilot.util

object HexStringUtil {
    fun hexToByteArray(value: String): ByteArray {
        return ByteArray(value.length / 2) { index ->
            value.substring(index * 2, index * 2 + 2).toInt(16).toByte()
        }
    }

    fun byteArrayToHex(value: ByteArray): String {
        val builder = StringBuilder()
        value.forEach { current ->
            builder.append(((current.toInt() shr 4) and 0xF).toString(16))
            builder.append((current.toInt() and 0xF).toString(16))
        }
        return builder.toString()
    }
}

